src/FacetedTokenInput.js

```diff
@@ -233,15 +258,30 @@ export default class FacetedTokenInput extends Component {
   }
 
   onChange(event) {
-    const { tokens } = this.state;
+    const { dir } = this.props;
+
+    const { tokens, dirChange, textDirection } = this.state;
 
     const searchText = this.refs.input.value;
 
+    let newTextDirection;
+
+    if (!dir) {
+      if (dirChange) {
+        newTextDirection = isRTL(searchText) ? 'rtl' : 'ltr';
+
+        this.setState({
+          dirChange: false,
+          textDirection: newTextDirection
+        });
+      }
+    }
+
```

Comment:
In this `if()` the direction of the text changes when it's not specified in the props (in this case the initial direction is set by your default language) and dirChange is set to false so the direction doesn't always try to change